app/controllers/targets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class TargetsController < ApplicationController
+  def new
+    @target = Target.new
+  end
+
+  def index
+    @targets = Target.all.last(10)
+  end
+
+  def create
+    @target = Target.create(target_params)
+    if @target.save
```

Comment:
when you do `Target.create(target_params)`, you are already saving the object in the database. I think you can check if the object was saved by doing `target.persisted?`. 

Another way to do this would be:

```
@target = Target.create!(target_params) # note that there is a bang (!) to raise an exception if fails to create
redirect_to targets_path
rescue ActiveRecord::RecordInvalid # The exception that is raised when an object fails to create due to some validation 
  flash.now[:alert] = 'Target was not saved'
end
```
So in this case if the target fails to be created, it will interrupt the flow of excecution and it will go directly to the rescue block